### PR TITLE
Restrict subnet auto-discovery to new LB creation only

### PR DESCRIPTION
**Pull request by oliviassss**
### Issue

#2056

### Description

Restrict the subnet auto-discovery to new load balancer creation on ingress side.

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/ingress/model_builder_test.go [hunk 1]
```diff
@@ -1835,34 +2114,578 @@ func Test_defaultModelBuilder_Build(t *testing.T) {
...
+    }
+}`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			elbv2Client := services.NewMockELBV2(ctrl)
```
Comment (kishorj) [inline pkg/ingress/model_builder_test.go:2659]: Why not use MockTaggingManager instead? The tests are simpler that way.

File: pkg/ingress/model_builder_test.go [hunk 2]
```diff
@@ -1835,34 +2114,578 @@ func Test_defaultModelBuilder_Build(t *testing.T) {
...
+			fields: fields{
+				describeLoadBalancersAsListCalls: []describeLoadBalancersAsListCall{
+					{
+						req: &elbv2sdk.DescribeLoadBalancersInput{},
+						resp: []*elbv2sdk.LoadBalancer{
+							{
+								LoadBalancerArn: awssdk.String("lb-1"),
+								AvailabilityZones: []*elbv2sdk.AvailabilityZone{
+									{
+										SubnetId: awssdk.String("subnet-a"),
```
[the remark this answers is not in the document]
Comment (oliviassss) [reply pkg/ingress/model_builder_test.go]: Changed to 'subnet-e' and 'subent-f' in this case.

File: pkg/ingress/model_builder_test.go (hunk 1 above)
Comment (oliviassss) [reply pkg/ingress/model_builder_test.go]: Thanks for your advice. Have changed to use MockTaggingManager.

**Pull request merged**